File: src/main/java/com/train/Ticket.java
package com.train;


import java.util.Scanner;

public class Ticket {


    private int ticketNum;
    private  int roundTrip;
    private static final double discount = 0.9;
    private static final int ticketPrice = 1000;
    private  boolean b = true;

    public Ticket() {
    }

    public Ticket(int ticketNum, int roundTrip) {
        this.ticketNum = ticketNum;
        this.roundTrip = roundTrip;

    }
    public void buyTicket(){


        do {


            Scanner scanner = new Scanner(System.in);
            System.out.println("Please enter number of tickets or enter -1 to end :");
            ticketNum = scanner.nextInt();

             if(ticketNum == -1){
                 b = false;
                 break;
            }
            System.out.println("How many round-trip tickets:");
            roundTrip = scanner.nextInt();

            if(roundTrip == -1 ){
                b = false;
                break;
            }
            if(roundTrip > ticketNum){
                System.out.println("來回票大於票數");
                break;
            }else{
                calculation();
            }


        }while (b);
        System.out.println("離開購票系列 謝謝!!");


    }
    /* 計算票券相關*/
    public void calculation(){
        int tickeCount = (ticketNum - roundTrip )*ticketPrice;

        double roundTripPrice = (roundTrip * discount)*ticketPrice*2;


        System.out.println("Total tickets: "+ ticketNum);
        System.out.println("Round-trip: "+ roundTrip);
        System.out.println("Total:" + (tickeCount + (int)roundTripPrice) );
    }

}

File: src/main/kotlin/com/train/Ticket.kt
package com.train.hd

import java.util.*


class Ticket(){
    val discount = 0.9
    val ticketPrice = 1000
    var tickNum :Int =0
    var roundtrip : Int = 0



    fun calculation(){

        var tickCount = (tickNum-roundtrip)*ticketPrice
        var roundCount = (roundtrip*discount)*ticketPrice*2
        println("Total tickets:" + tickNum)
        println("Round-trip:"+roundtrip)
        println("Total: "+(tickCount+roundCount))
    }

    fun buyTickets() {
        var b = true
        do{
            var scanner = Scanner(System.`in`)
            println("Please enter number of tickets or enter -1 to end :")
            tickNum = scanner.nextInt()
            if(tickNum == -1){
                b = false
                break
            }
            println("How many round-trip tickets:")
            roundtrip = scanner.nextInt()
            if(roundtrip == -1){
                b = false
                break
            }
            if(roundtrip > tickNum){
                println("來回票大於票數")
                break
            }
            calculation()
        }while (b)
      print("離開購票系列 謝謝!!")
    }


}